### PR TITLE
Refactoring: Convert plugin event calls to foreach style in main App class

### PR DESCRIPTION
**Pull request by makiam**
just a code clean

### CODE REVIEW

File: ArtOfIllusion/src/artofillusion/ArtOfIllusion.java [hunk 1]
```diff
@@ -371,51 +367,50 @@ public static void quit()
...
     System.exit(0);
   }
 
   /** Execute all startup scripts. */
-
+  @SuppressWarnings("CallToPrintStackTrace")
   private static void runStartupScripts()
   {
     String files[] = new File(STARTUP_SCRIPT_DIRECTORY).list();
+    if(null == files) return;
```
Comment (peastman) [inline ArtOfIllusion/src/artofillusion/ArtOfIllusion.java]: I know it's just nitpicking, but can we maintain a consistent code style?  I think that helps readability.  That should be

```Java
if (null == files)
  return;
```

File: ArtOfIllusion/src/artofillusion/ArtOfIllusion.java [hunk 2]
```diff
@@ -1,4 +1,5 @@
 /* Copyright (C) 1999-2013 by Peter Eastman
+   Copyright (C) 2016 by Maksim Khramov
```
Comment (Sailsman63) [inline ArtOfIllusion/src/artofillusion/ArtOfIllusion.java]: For clarity, I'd suggest that Copyright notifications for edits should start "Changes Copyright..." as your copyright only applies to the code that you've added or changed. I know it's another nitpick, but better to get things figured out before it gets us in trouble...

File: ArtOfIllusion/src/artofillusion/ArtOfIllusion.java [hunk 3]
```diff
@@ -272,7 +273,7 @@ public static void newWindow()
   }
 
   /** Create a new window for editing the specified scene. */
-
+  @SuppressWarnings("CallToPrintStackTrace")
```
Comment (Sailsman63) [inline ArtOfIllusion/src/artofillusion/ArtOfIllusion.java]: Another thing I was wondering about: @peastman, do we want to get into the habit of suppressing compiler warnings?

Comment (peastman) [inline ArtOfIllusion/src/artofillusion/ArtOfIllusion.java]: If you're really certain the warning is inappropriate, it's probably a reasonable thing to do.

Comment (Sailsman63) [inline ArtOfIllusion/src/artofillusion/ArtOfIllusion.java]: And that is the core of my question. I know that we tabled the discussion about using a logger for the time being. Currently, a lot of AOI code uses printStackTrace() at the time an exception occurs. If we want this to be the "Official" standard way to handle exceptions, we should run through the entire codebase, suppress this warning on any method that contains a `catch` block, and then modify the launchers to redirect stderr to a file.

If we decide to do something different for logging, the warnings are good in that they remind us of where some of our legacy code lives. Either way, we should be consistant.

**Comment by Sailsman63** [comment]
Thanks! Merged.

**Pull request merged**